### ISSUE TITLE
Using k8s 1.9.7+ produces clusters that do work work and have missing kube-system pods

### ISSUE THREAD

**Issue by karlmutch** [Azure/acs-engine#2828]
**Is this a request for help?**:
Yes
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Issue
---

**What version of acs-engine?**:
Version: v0.16.1
GitCommit: 232907b97
GitTreeState: clean
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
k8s 1.9.7, 1.10.2

**What happened**:
Using base acs-engine results in a cluster where only 2 kube-system pods are ever successfully scheduled (apiserver, controller-manager).  When using k8s 1.8.11 everything works and the full complement of system pods run.

With k8s 1.9.7 the apiserver pods have multiple restarts

On occasion we have a stack trace
E0501 17:53:44.205844       1 status.go:64] apiserver received an error that is not an metav1.Status: Get https://k8s-master-32591565-0:10250/
containerLogs/kube-system/kube-apiserver-k8s-master-32591565-0/kube-apiserver?tailLines=50: dial tcp: lookup k8s-master-32591565-0 on 168.63.1
29.16:53: no such host
I0501 17:53:44.206587       1 wrap.go:42] GET /api/v1/namespaces/kube-system/pods/kube-apiserver-k8s-master-32591565-0/log?container=kube-apis
erver&tailLines=50: (6.855354ms) 500
goroutine 12604 [running]:
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc4285c9b90, 0x1f4)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/httplog/httplog.go:207 +0xdd
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc4285c9b90, 0x1f4)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/httplog/httplog.go:186 +0x35
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.(*auditResponseWriter).WriteHeader(0xc429dddb30, 0x1f4)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/filters/audit.go:220 +0x63
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics.(*ResponseWriterDelegator).WriteHeader(0xc426f56ae0, 0x1f4)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/metrics/metrics.go:258 +0x45
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.SerializeObject(0x49f8bfb, 0x10, 0x7efe66bdc030, 0xc4244b7780
, 0xa742a40, 0xc428e5d558, 0xc429350d00, 0x1f4, 0xa723680, 0xc428946b40)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/responsewriters/writers.go:95 +0x8d
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObjectNegotiated(0x7efe61f7af68, 0xc426f56bd0, 0xa746680
, 0xc420e73b60, 0x0, 0x0, 0x49d80b8, 0x2, 0xa742a40, 0xc428e5d558, ...)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/responsewriters/writers.go:123 +0x32b
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.ErrorNegotiated(0x7efe61f7af68, 0xc426f56bd0, 0xa710900, 0xc4
26f57bc0, 0xa746680, 0xc420e73b60, 0x0, 0x0, 0x49d80b8, 0x2, ...)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/responsewriters/writers.go:142 +0x163
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.StreamObject(0x7efe61f7af68, 0xc426f56bd0, 0xc8, 0x0, 0x0, 0x
49d80b8, 0x2, 0xa746680, 0xc420e73b60, 0x7efe6158c4d0, ...)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/responsewriters/writers.go:61 +0x702
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObject.func1()
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/responsewriters/writers.go:48 +0xa3
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics.RecordLongRunning(0xc429350d00, 0xc4293ca000, 0xc42d8df770)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/metrics/metrics.go:155 +0x1c8
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters.WriteObject(0x7efe61f7af68, 0xc426f56bd0, 0xc8, 0x0, 0x0, 0x4
9d80b8, 0x2, 0xa746680, 0xc420e73b60, 0xa724400, ...)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/responsewriters/writers.go:47 +0x1cf
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.transformResponseObject(0x7efe61f7af68, 0xc426f56bd0, 0xa7542a0, 0xc421461780
, 0xc4208a3680, 0xa746680, 0xc420e73b60, 0xa723a80, 0xc420389300, 0xa70d580, ...)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/response.go:172 +0x26bc
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/handlers.getResourceHandler.func1(0xa742a40, 0xc428e5d558, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/handlers/get.go:76 +0x3ba
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints.restfulGetResourceWithOptions.func1(0xc426f569f0, 0xc4293ab4a0)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/installer.go:1116 +0xb0
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/metrics.InstrumentRouteFunc.func1(0xc426f569f0, 0xc4293ab4a0)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/metrics/metrics.go:199 +0x208
k8s.io/kubernetes/vendor/github.com/emicklei/go-restful.(*Container).dispatch(0xc42159fcb0, 0xa742940, 0xc428e5d548, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/
emicklei/go-restful/container.go:277 +0xb18
k8s.io/kubernetes/vendor/github.com/emicklei/go-restful.(*Container).Dispatch(0xc42159fcb0, 0xa742940, 0xc428e5d548, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/
emicklei/go-restful/container.go:199 +0x57
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x49f1a00, 0xe, 0xc42159fcb0, 0xc4205237a0, 0xa742940, 0xc428e5d548, 0
xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/handler.go:153 +0x4e0
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc420b538c0, 0xa742940, 0xc428e5d548, 0xc429350d00)
        <autogenerated>:1 +0x75
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver.(*proxyHandler).ServeHTTP(0xc429450a80, 0xa742940, 0xc428e5d548, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kube
-aggregator/pkg/apiserver/handler_proxy.go:91 +0x15f
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc425b3cc80, 0xa742940, 0xc428e5d548, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/mux/pathrecorder.go:248 +0x26d
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc4201a0620, 0xa742940, 0xc428e5d548, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/mux/pathrecorder.go:234 +0xa1
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x49f1a00, 0xe, 0xc42159fcb0, 0xc4205237a0, 0xa742940, 0xc428e5d548, 0
xc429350d00)                                                                                                                                  
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/handler.go:153 +0x4e0                                                                                                        
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc420b538c0, 0xa742940, 0xc428e5d548, 0xc429350d00)               
        <autogenerated>:1 +0x75                                                                                                               
k8s.io/kubernetes/vendor/k8s.io/kube-aggregator/pkg/apiserver.(*proxyHandler).ServeHTTP(0xc429450a80, 0xa742940, 0xc428e5d548, 0xc429350d00)  
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kube
-aggregator/pkg/apiserver/handler_proxy.go:91 +0x15f                                                                                          
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc425b3cc80, 0xa742940, 0xc428e5d548, 0xc429350d00)        
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/mux/pathrecorder.go:248 +0x26d                                                                                               
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc4201a0620, 0xa742940, 0xc428e5d548, 0xc429350d00)    
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/mux/pathrecorder.go:234 +0xa1                                                                                                
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x49f5573, 0xf, 0xc424213170, 0xc4201a0620, 0xa742940, 0xc428e5d548, 0
xc429350d00)                                                                                                                                  
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/handler.go:161 +0x6ad                                                                                                        
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*director).ServeHTTP(0xc423fe7380, 0xa742940, 0xc428e5d548, 0xc429350d00)               
        <autogenerated>:1 +0x75                                                                                                               
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0xa742940, 0xc428e5d548, 0xc429350d00)                
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/filters/authorization.go:51 +0x37d                                                                                        
net/http.HandlerFunc.ServeHTTP(0xc4226e5450, 0xa742940, 0xc428e5d548, 0xc429350d00)                                                           
        /usr/local/go/src/net/http/server.go:1918 +0x44                                                                                       
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0xa742940, 0xc428e5d548, 0xc429350d00)                
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/filters/maxinflight.go:131 +0x515                                                                                            
net/http.HandlerFunc.ServeHTTP(0xc4216a2680, 0xa742940, 0xc428e5d548, 0xc429350d00)                                                           
        /usr/local/go/src/net/http/server.go:1918 +0x44                                                                                       
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0xa742940, 0xc428e5d548, 0xc429350d00)                
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/filters/impersonation.go:49 +0x203a                                                                                       
net/http.HandlerFunc.ServeHTTP(0xc4226e54a0, 0xa742940, 0xc428e5d548, 0xc429350d00)                                                           
        /usr/local/go/src/net/http/server.go:1918 +0x44                                                                                       
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAudit.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)                        
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/filters/audit.go:105 +0x501                                                                                               
net/http.HandlerFunc.ServeHTTP(0xc4226e5540, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/filters/authentication.go:79 +0x2b1
net/http.HandlerFunc.ServeHTTP(0xc4226e55e0, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.WithRequestContext.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/request/requestcontext.go:110 +0xcb
net/http.HandlerFunc.ServeHTTP(0xc423fe73a0, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP(0xc423fe7420, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/filters/timeout.go:86 +0x422
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithWaitGroup.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/filters/waitgroup.go:51 +0xeb
net/http.HandlerFunc.ServeHTTP(0xc4216a2740, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithRequestInfo.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/filters/requestinfo.go:45 +0x20f
net/http.HandlerFunc.ServeHTTP(0xc4216a2880, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/request.WithRequestContext.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/endpoints/request/requestcontext.go:110 +0xcb
net/http.HandlerFunc.ServeHTTP(0xc423fe7440, 0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithPanicRecovery.func1(0xa742ac0, 0xc4285c9b90, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/filters/wrap.go:41 +0x108
net/http.HandlerFunc.ServeHTTP(0xc423fe7480, 0xa743f40, 0xc4205689c8, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:1918 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.(*APIServerHandler).ServeHTTP(0xc4240edf50, 0xa743f40, 0xc4205689c8, 0xc429350d00)
        /workspace/anago-v1.9.7-beta.0.111+dd5e1a2978fd0b/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apis
erver/pkg/server/handler.go:198 +0x51
net/http.serverHandler.ServeHTTP(0xc424200ea0, 0xa743f40, 0xc4205689c8, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:2619 +0xb4
net/http.initNPNRequest.ServeHTTP(0xc42449b500, 0xc424200ea0, 0xa743f40, 0xc4205689c8, 0xc429350d00)
        /usr/local/go/src/net/http/server.go:3164 +0x9a
net/http.(*initNPNRequest).ServeHTTP(0xc4264f3490, 0xa743f40, 0xc4205689c8, 0xc429350d00)
        <autogenerated>:1 +0x63
net/http.(Handler).ServeHTTP-fm(0xa743f40, 0xc4205689c8, 0xc429350d00)
        /usr/local/go/src/net/http/h2_bundle.go:5462 +0x4d
net/http.(*http2serverConn).runHandler(0xc42531ee00, 0xc4205689c8, 0xc429350d00, 0xc4255e8d00)
        /usr/local/go/src/net/http/h2_bundle.go:5747 +0x89
created by net/http.(*http2serverConn).processHeaders
        /usr/local/go/src/net/http/h2_bundle.go:5481 +0x495

logging error output: "{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"Get https://k8s-master-
32591565-0:10250/containerLogs/kube-system/kube-apiserver-k8s-master-32591565-0/kube-apiserver?tailLines=50: dial tcp: lookup k8s-master-32591
565-0 on 168.63.129.16:53: no such host\",\"code\":500}\n"
 [[kk/v0.0.0 (linux/amd64) kubernetes/$Format] 34.218.195.166:59360]
I0501 17:53:44.806503       1 wrap.go:42] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.76254ms) 200 [[hyperkube/v1
.9.7 (linux/amd64) kubernetes/dd5e1a2/leader-election] 10.255.255.5:34670]

All other services and deployments I try to deploy stall and are never observed.

**What you expected to happen**:

I expected a cluster on which the standard pods are running.

**How to reproduce it** (as minimally and precisely as possible):

Use acs-engine to deploy a single node cluster using 1.9+ k8s and the most recent version of acs-engine

**Anything else we need to know**:

This also occurs with the latest k8s 1.10

**Comment by jackfrancis** [comment]
Hi @karlmutch can you share (without credentials) the original api model used to deploy the cluster? We can try to repro.

**Comment by karlmutch** [comment]
File below, you'll need to use a region that supports the GPU instances:

{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.9"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "",
      "vmSize": "Standard_D2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool1",
        "count": 1,
        "vmSize": "Standard_ND6s",
        "OSDiskSizeGB": 200,
        "availabilityProfile": "AvailabilitySet"
      }
    ],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "*****",
      "secret": "****"
    }
  }
}

**Comment by jackfrancis** [comment]
@wbuchwalter @sozercan Does current GPU support in acs-engine include versions of k8s >= 1.9? Our E2E tests validate 1.8.

**Comment by wbuchwalter** [comment]
@jackfrancis Should work with anything < 1.11 as far as I know.

**Comment by sam-cogan** [comment]
I'm also seeing this with 1.9 and 1.10 clusters. I only get the APIServer and controller-manger pods running. Template is below, not doing much special other than mixed windows and linux.
```

{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.9"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "acs-engine-mixed",
      "vmSize": "Standard_D2_v3",
      "OSDiskSizeGB": 64,
      "storageProfile" : "ManagedDisks"
    },
    "agentPoolProfiles": [
      {
          "name": "linuxpool",
          "count": 1,
          "vmSize": "Standard_D2_v3",
          "availabilityProfile": "AvailabilitySet",
          "OSDiskSizeGB": 64,
          "storageProfile" : "ManagedDisks"
      },
      {
          "name": "windowspool",
          "count": 1,
          "vmSize": "Standard_D2_v3",
          "availabilityProfile": "AvailabilitySet",
          "OSDiskSizeGB": 128,
          "storageProfile" : "ManagedDisks",
          "osType": "Windows"
      }
    ],
    "windowsProfile": {
      "adminUsername": "",
      "adminPassword": ""
    },
    "linuxProfile": {
      "adminUsername": "",
      "ssh": {
        "publicKeys": [
          {
            "keyData": ""
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "",
      "secret": ""
    }
  }
}
```

**Comment by karlmutch** [comment]
Thanks, I believe that we are seeing the same cause.  Possibly the only action to come from this might be for an improvement that displays an error message rather than a full stack trace.

**Comment by sam-cogan** [comment]
Yep, that looks like the reason, works fine for me now permissions are correct.

**Comment by karlmutch** [comment]
@jackfrancis If the only action from this is that there could be an improvement to logging to make it apparent that a permissions issue exists when creating the cluster, should I close this, or reword this as a request to improve logging when permissions problems exist for service principals ?
